src/kit.ts

```diff
@@ -755,7 +834,7 @@ export async function getVSKitEnvironment(kit: Kit): Promise<Map<string, string>
 }
 
 export async function effectiveKitEnvironment(kit: Kit, opts?: expand.ExpansionOptions): Promise<Map<string, string>> {
-  const host_env = objectPairs(process.env) as [string, string][];
+  let host_env = objectPairs(process.env) as [string, string][];
```

Comment:
`host_env` shouldn't be populated here if you're going to throw the result away when an environment script runs.

Reply:
Okay, I changed that so host_env only becomes populated if the shell / setup script was not run successfully.